feg/gateway/services/n7_n40_proxy/servicers/create_session.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2021 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package servicers
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/golang/glog"
+
+	n7_sbi "magma/feg/gateway/sbi/specs/TS29512NpcfSMPolicyControl"
+	"magma/feg/gateway/services/n7_n40_proxy/n7"
+	"magma/lte/cloud/go/protos"
+)
+
+// getSmPolicyRulesIfEnabled sends the request to PCF and get the response when N7 is enabled,
+// otherwise returns an empty SmPolicyDecision
+// Returns SmPolicyDecision rules and smPolicyUrl that uniquely identifies the policy session.
+func (srv *CentralSessionController) getSmPolicyRulesIfEnabled(
+	request *protos.CreateSessionRequest,
+) (*n7_sbi.SmPolicyDecision, string, error) {
+	if srv.cfg.N7Config.DisableN7 {
+		// Empty response when disabled
+		return &n7_sbi.SmPolicyDecision{
+			PccRules: &n7_sbi.SmPolicyDecision_PccRules{},
+		}, "", nil
+	}
+	// Convert and send the request to PCF
+	reqBody := n7.GetSmPolicyContextDataN7(request, srv.cfg.N7Config.Client.NotifyApiRoot)
+	reqCtx, cancel := context.WithTimeout(context.Background(), srv.cfg.RequestTimeout)
+	defer cancel()
+	resp, err := srv.policyClient.PostSmPoliciesWithResponse(reqCtx, *reqBody)
+	if err != nil {
+		return nil, "", err
+	}
+	if resp.StatusCode() != http.StatusOK && resp.StatusCode() != http.StatusCreated {
+		err = fmt.Errorf("SmPolicyCreate request failure: status-code=%d", resp.StatusCode())
+		return nil, "", err
+	}
+	policyUrl, found := resp.HTTPResponse.Header["Location"]
+	if !found {
+		err = fmt.Errorf("SmPolicyCreate request failure: Location header not found")
+		return nil, "", err
+	}
+	return resp.JSON201, policyUrl[0], nil
+}
+
+func (srv *CentralSessionController) injectOmnipresentRules(
+	policy *n7_sbi.SmPolicyDecision,
+) error {
```

Comment:
try to use one line function if it fits.
Note when you do build.py -c it will break the line if necessary

Reply:
Why is tis function returning error?  I don't see any being returned